integration/generic-tests/kexec_upl_test.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2024 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build !race
+// +build !race
+
+package integration
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/hugelgupf/vmtest/qemu"
+	"github.com/hugelgupf/vmtest/scriptvm"
+	"github.com/hugelgupf/vmtest/testtmp"
+	"github.com/u-root/mkuimage/uimage"
+)
+
+func TestKexecUpl(t *testing.T) {
+	qemu.SkipIfNotArch(t, qemu.ArchAMD64, qemu.ArchArm64)
+
+	script := `
+		kexec /upl
+	`
+
+	initrd := filepath.Join(testtmp.TempDir(t), "initramfs.cpio")
+	vm := scriptvm.Start(t, "vm", script,
+		scriptvm.WithUimage(
+			// Build kexec as a binary command to get accurate GOCOVERDIR
+			// integration coverage data (busybox rewrites command code).
+			uimage.WithCoveredCommands("github.com/u-root/u-root/cmds/core/kexec"),
+			uimage.WithFiles(fmt.Sprintf("%s:upl", os.Getenv("VMTEST_UPL"))),
+			uimage.WithCPIOOutput(initrd),
+		),
+		scriptvm.WithQEMUFn(
+			qemu.WithVMTimeout(time.Minute),
+			// Specify the machine type to report correct ACPI RSDP table
+			qemu.ArbitraryArgs("-M", "q35"),
+			qemu.ArbitraryArgs("-m", "8192"),
+			// EDK2 requries processor to support TSC / Core Crystal ratio feature
+			// Specify the processor type to meet abot EDK2 requirment.
+			qemu.ArbitraryArgs("-cpu", "Skylake-Client"),
+			// Enable KVM to speed up test. However, KVM kernel module might not
+			// be installed and enabled by default. Remove this line it host OS
+			// is lack of KVM support
+			qemu.ArbitraryArgs("-enable-kvm"),
```

Comment:
Our automated test environment (Github Actions) unfortunately I *think* won't support KVM. Locally, I tend to run `VMTEST_QEMU_APPEND="-enable-kvm" runvmtest -- go test ./...` to speed up tests.

See https://github.com/hugelgupf/vmtest?tab=readme-ov-file#running-tests

Reply:
Will try it later.

Reply:
Removed the arg.